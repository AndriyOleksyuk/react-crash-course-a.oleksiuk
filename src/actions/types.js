const types = {
    START_FETCH_DATA: 'START_FETCH_DATA',
    FINISH_FETCH_DATA: 'FINISH_FETCH_DATA',
    HANDLE_ERROR: 'HANDLE_ERROR',
    SET_BUTTONS_SIZE: 'SET_BUTTONS_SIZE',
    SET_BUTTONS_OUTLINE: 'SET_BUTTONS_OUTLINE',
    SET_CANCEL_TOKEN: 'SET_CANCEL_TOKEN'
}

export default types;