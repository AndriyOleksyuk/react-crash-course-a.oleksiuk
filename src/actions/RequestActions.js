import axios from 'axios';
import types from './Types';

export const cancelRequest = () => (dispatch, getState) => {
    let { source } = getState().request;
    const CancelToken = axios.CancelToken;

    source.cancel('Operation canceled by the user.');
   
    source = CancelToken.source();
    dispatch(setCancelSource(source));
}

export const fetchUserData = () => (dispatch, getState) => {
    let { source } = getState().request;
    const URL = 'https://randomuser.me/api/';
    const cancelToken = source.token;

    dispatch(startFetchUserData());

    return axios.get(URL, {cancelToken})
        .then(res => res.data)
        .then(function (data) {
            var userInfo = data.results[0];            
            dispatch(finishFetchUserData({email: userInfo.email, data: JSON.stringify(data, null, 4)}));
        }).catch(function (thrown) {
            if (axios.isCancel(thrown)) {
                dispatch(finishFetchUserData({}));
            } else {
                dispatch(handleError(thrown.message));
            }
        });
};

export const setCancelSource = (source) => ({ type: types.SET_CANCEL_TOKEN, source});
export const startFetchUserData = () => ({ type: types.START_FETCH_DATA});
export const finishFetchUserData = (response) => ({ type: types.FINISH_FETCH_DATA, response});
export const handleError = (error) => ({ type: types.HANDLE_ERROR, error});