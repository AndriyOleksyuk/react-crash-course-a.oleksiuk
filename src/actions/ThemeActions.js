
import types from './Types';

export const setBtnSize = (size) => ({type: types.SET_BUTTONS_SIZE, size});
export const setBtnOutline = (outlineValue) => ({type: types.SET_BUTTONS_OUTLINE, outlineValue});