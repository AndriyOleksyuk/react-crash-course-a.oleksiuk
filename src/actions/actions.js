import axios from 'axios';
import types from './types';

export const cancelRequest = (source) => (dispatch) => {
    source.cancel('Operation canceled by the user.');

    const CancelToken = axios.CancelToken;
    source = CancelToken.source();
    dispatch(setCancelSource(source));
}

export const fetchUserData = (source) => (dispatch) => {
    const URL = 'https://randomuser.me/api/';
    const cancelToken = source.token;

    dispatch(startFetchUserData());

    axios.get(URL, {cancelToken})
        .then(res => res.data)
        .then(function (data) {
            var userInfo = data.results[0];            
            dispatch(finishFetchUserData({email: userInfo.email, data: JSON.stringify(data, null, 4)}));
        }).catch(function (thrown) {
            if (axios.isCancel(thrown)) {
                dispatch(finishFetchUserData({}));
            } else {
                dispatch(handleError(thrown.message));
            }
        });
};

export const setCancelSource = (source) => ({ type: types.SET_CANCEL_TOKEN, source});
export const startFetchUserData = () => ({ type: types.START_FETCH_DATA});
export const finishFetchUserData = (response) => ({ type: types.FINISH_FETCH_DATA, response});
export const handleError = (error) => ({ type: types.HANDLE_ERROR, error});
export const setBtnSize = (size) => ({type: types.SET_BUTTONS_SIZE, size});
export const setBtnOutline = (outlineValue) => ({type: types.SET_BUTTONS_OUTLINE, outlineValue});