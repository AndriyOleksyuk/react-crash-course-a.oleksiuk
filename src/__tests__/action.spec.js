import { setBtnSize, setBtnOutline } from '../actions/ThemeActions';

describe('actions', () => {
    it('should set btn size', () => {
        const action = setBtnSize('sm');
        expect(action).toMatchSnapshot();
    })

    it('should set outline', () => {
        const action = setBtnOutline('outline-');
        expect(action).toMatchSnapshot();
    })

    it('should set delete outline', () => {
        const action = setBtnOutline('');
        expect(action).toMatchSnapshot();
    })
})
