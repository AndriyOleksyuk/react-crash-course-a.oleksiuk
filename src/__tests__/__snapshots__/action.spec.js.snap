// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should set btn size 1`] = `
Object {
  "size": "sm",
  "type": "SET_BUTTONS_SIZE",
}
`;

exports[`actions should set delete outline 1`] = `
Object {
  "outlineValue": "",
  "type": "SET_BUTTONS_OUTLINE",
}
`;

exports[`actions should set outline 1`] = `
Object {
  "outlineValue": "outline-",
  "type": "SET_BUTTONS_OUTLINE",
}
`;
